### DISCUSSION TITLE
关于自建测试地址问题咨询~

### DISCUSSION THREAD

**Discussion by wulong2020** [XIU2/CloudflareSpeedTest#458] [讨论]
按照大佬 之前在#168给出的自建测速地址方案二：
2. 服务器套 Cloudflare CDN（需要服务器+域名）

我在AWS EC2上创建了一个网站启用了NGINX，同时放了测速文件在主目录下面，然后我套用的是AWS 家的CFT ，效果很好。

![Snipaste_2023-11-10_11-18-02](https://github.com/XIU2/CloudflareSpeedTest/assets/86549530/c00b588d-2f34-4598-9782-c4004d7f8767)

目前我的疑问是：

因为源站测速文件是放在AWS EC2上面的，如果经常发起CDN优选IP测速，CFT的流量会消耗这个我明白，**请问会不会也会消耗EC2（源站）的流量，因为AWS EC2每个月免费就100G的流量**

**Comment by charSLee013** [comment]
可以用 `HEAD` 访问去查看是否命中缓存，比如说一个steam的视频文件

```shell
❯ curl -I 'https://cdn.cloudflare.steamstatic.com/steam/apps/256843155/movie_max.mp4'
HTTP/1.1 200 Connection established

HTTP/2 200
date: Fri, 10 Nov 2023 08:39:03 GMT
content-type: video/mp4
content-length: 213013221
last-modified: Fri, 16 Jul 2021 03:36:13 GMT
cf-cache-status: HIT   <-- 表示命中缓存
age: 1055
accept-ranges: bytes
server: cloudflare
```

如果没有命中则可能需要调整 `cloudflare cache` 相关规则了

**Reply by wulong2020** [reply]
<img width="910" alt="Snipaste_2023-11-10_20-28-54" src="https://github.com/XIU2/CloudflareSpeedTest/assets/86549530/2006843a-3a35-408f-a8cd-2c073442c50e">
请问我这个是被CDN缓存了么？如果CDN缓存了，应该不用去源站发请求了，那这个CDN缓存会自己清理掉么

**Reply by charSLee013** [reply]
缓存了，不过cloudfront 是AWS的，说明你这个文件被AWS缓存了而不是cloudflare

到cloduflare 面板DNS 选项里面，检查一下是不是没有开启橙色云朵
<img width="214" alt="image" src="https://github.com/XIU2/CloudflareSpeedTest/assets/20462328/89bc851f-8100-4103-acaf-985895cacb15">

或者去缓存 - Tiered Cache 中开启 Argo Tiered Cache 功能后，再度检查是不是有缓存

**Reply by wulong2020** [reply]
谢谢回复，因为我自建测速节点就是套用的AWS的CFT，CF家的CDN太慢了

**Reply by wulong2020** [reply]
另外，请问这个问题能解答下么：

因为源站测速文件是放在AWS EC2上面的，如果经常发起CDN优选IP测速，CFT的流量会消耗这个我明白，请问会不会也会消耗EC2（源站）的流量，因为AWS EC2每个月免费就100G的流量

**Reply by charSLee013** [reply]
好问题我也不知道因为我没用AWS家服务

一般来讲缓存流量和回源流量是分开计算的，比如<AWS EC2每个月免费100G的流量>就是算回源和流出流量

AWS应该有CDN面板可以查看当前缓存流量和回源流量用的是多少，以下面的cloudflare 面板来说明的话

<img width="490" alt="image" src="https://github.com/XIU2/CloudflareSpeedTest/assets/20462328/e2512616-9437-49c0-bf18-aadcf5e0c0a5">

用 **Data Served** - **Data Cached** = 回源流量

**Reply by wulong2020** [reply]
感谢一起探讨，AWS家的计费方式出了名了难懂，如果按照AWS官方说的这个说法

<img width="1416" alt="Snipaste_2023-11-11_10-09-36" src="https://github.com/XIU2/CloudflareSpeedTest/assets/86549530/48acccbe-59df-47db-8467-a276e131fb1b">

那使用AWS家的CFT区访问自家的EC2（源站）测速，应该是免费的

不过，该说不说，CFT的速度确实比CF快的不是一点两点

**Reply by wulong2020** [reply]
<img width="1994" alt="Snipaste_2023-11-11_10-21-15" src="https://github.com/XIU2/CloudflareSpeedTest/assets/86549530/2eed1601-cbd5-4a6f-8500-5567b61a5b5d">

到CFT账单仔细看了下，CFT和EC2之间数据传输应该是免费的，这样自建的EC2测速地址，套CFT，应该不用考虑EC2 流量计费了

**Reply by redcats2** [reply]
好像《AWS EC2每个月免费100G的流量》并不包含回源和跨区流量，仅为本区流出，回源和跨区是单另计算的，跨区每个月免费1G，超过收费，回源没有免费，亚太大概1G 7美分左右

**Comment by soapmancn** [comment]
为什么你这个测速延迟这么低 是付费用户吗 我测速全是140几延迟
